### DISCUSSION TITLE
`jj resolve` marks file as resolved despite containing conflict markers

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#4877] [Q&A] [answered]
I'm using `vscode` as my diff editor, and I've noticed that if I close the window without making any changes, the file is considered "resolved" by `jj status`, even though there are still conflict markers. I don't understand exactly what's happening, and I think I remember this not always being the case (i.e. when closing the window without resolving the conflicts, `jj` didn't update the conflict status). Unfortunately, I don't remember when it changed.

What causes `jj` to decide that a conflict is resolved? Is it possible this is a change on the VSCode side? I've disabled the warning on close that says "there are conflicts remaining", so did that maybe change the behavior?

**Comment by martinvonz** [comment, marked as answer]
See https://martinvonz.github.io/jj/latest/config/#editing-conflict-markers-with-a-tool-or-a-text-editor.  We have `merge-tool-edits-conflict-markers = true` set for VS Code [here[(https://github.com/martinvonz/jj/blob/1ed2ef04058f19c055a03e3a3dea2b932d0bfd91/cli/src/config/merge_tools.toml#L54), so jj shouldn't consider the file resolved. Maybe you modified the conflict marker a bit so it couldn't be parsed?

**Reply by BatmanAoD** [reply]
I'm sorry, I think I may have just been misreading the diff, and it was showing that the conflict markers were in fact deleted. It does appear that if I resolve any changes in VSCode it deletes all remaining conflict markers, though, so I need to figure out how to fix that configuration issue...

**Reply by ilyagr** [reply]
I have not found a way to prevent VS Code from doing that. It seems necessary to resolve the conflict in one step. Let me know if you have more success.

**Comment by ilyagr** [comment]
+1 to Martin.

Are the conflict markers still `jj`'s conflict markers or do they look different? I'm wondering whether VS Code overwrote the file with its own understanding of the remaining conflicts, or messed them up like Martin suggested. I seem to remember VS Code didn't handle the situation of partially resolved conflicts well, but it could also have gotten worse (or better) since I last checked.
